### ISSUE TITLE
Customizing document title used in index

### ISSUE THREAD

**Issue by stefanseefeld** [Polyconseil/dokang#18]
It appears as if the top-level document title, which is shown in the index, corresponds to the title of the `index.html` file in the uploaded docs. Is there a way to customize that, i.e. use a different title ? For example, in sphinx auto-generated docs the default title always starts with "Welcome to ...", so the dokang index would yield a list of titles all starting with "Welcome to ...", which isn't very useful.

Thanks,

**Comment by dbaty** [comment]
Good point.  (Although this is a good sign that your documentation writers should perhaps be more creative with their title? ;) )

I think that we should be able to provide a custom title when uploading the documentation. From a (very quick!) glance, that would mean:

- adding  a `title` field in the [upload form around here](https://github.com/Polyconseil/dokang/blob/e62088c1dc5b1c77d204ccedf26f613fb8d57873/src/dokang/views.py#L115-L117). This field should be optional for backward compatibility;
- passing this `title` to `utils.doc_set()` when [called here](https://github.com/Polyconseil/dokang/blob/e62088c1dc5b1c77d204ccedf26f613fb8d57873/src/dokang/views.py#L170);
- adding a `title=None` argument to [utils.doc_set()](https://github.com/Polyconseil/dokang/blob/e62088c1dc5b1c77d204ccedf26f613fb8d57873/src/dokang/utils.py#L15);
- in `utils.doc_set()`, remove [this line](https://github.com/Polyconseil/dokang/blob/e62088c1dc5b1c77d204ccedf26f613fb8d57873/src/dokang/utils.py#L21C3-L21C3) so that, by default, the title is what we send to the upload form.

As you may have seen from the commit history, there has not been any substantial contribution to Dokang for a while. I personally don't use Dokang anymore, but I would be willing to review a pull request if you feel inclined to write one. I don't see any test for the upload feature: if you could add one, that would be great.

**Comment by stefanseefeld** [comment]
:-)

But creativity questions aside, I can see situations where the toplevel page title is not what I want to see in the index.
Thanks for the suggestions, I'll be working on a PR to add an (optional) title argument.
